src/renderer/src/components/TransactionActivityList/index.tsx

```diff
@@ -33,15 +36,43 @@ const heights = {
 
 export const TransactionActivityList = ({ defaultAccounts }: TProps) => {
   const { t } = useTranslation('components', { keyPrefix: 'transactionActivityList' })
-  const { actionData } = useActions<TActionsData>({ accounts: defaultAccounts })
 
   const dateNow = new Date()
-  const dateFrom = startOfMonth(dateNow).toISOString()
-  const dateTo = dateNow.toISOString()
 
-  const { data, isLoading, fetchNextPage } = useGetFullTransactions({ ...actionData, dateFrom, dateTo })
+  const { actionData, setData } = useActions<TActionsData>({
+    accounts: defaultAccounts,
+    dateFrom: dateFns.startOfMonth(dateNow),
+    dateTo: dateNow,
+  })
+
+  const { data, isLoading, fetchNextPage } = useGetFullTransactions(actionData)
   const { handleScroll, ref: scrollRef } = useInfiniteScroll<HTMLDivElement>(fetchNextPage)
 
+  const { dateFrom, dateTo } = actionData
+
+  const isDateDisabled = isLoading ? () => true : { after: dateNow }
+
+  const handleSelectDateFrom = (date?: Date) => {
+    if (!date) return
+
+    setData({ dateFrom: date })
+
+    if (dateTo && dateFns.isAfter(date, dateTo))
+      setData({ dateTo: dateFns.min([dateNow, dateFns.add(date, { weeks: 1 })]) })
+    else if (dateTo && dateFns.differenceInYears(dateTo, date) > 0)
+      setData({ dateTo: dateFns.add(date, { years: 1, days: -1 }) })
```

Comment:
Instead of using an if else, use if with a return, this way the code is less indented and easier to read.

```suggestion
    if (dateTo && dateFns.isAfter(date, dateTo)) {
      setData({ dateTo: dateFns.min([dateNow, dateFns.add(date, { weeks: 1 })]) })
      return
    }
    
    if (dateTo && dateFns.differenceInYears(dateTo, date) > 0) {
      setData({ dateTo: dateFns.add(date, { years: 1, days: -1 }) })
    }
```